### DISCUSSION TITLE
Will there be performance issues running WSL2 on an Intel big.LITTLE CPU ?

### DISCUSSION THREAD

**Discussion by dsyx** [microsoft/WSL#9190] [General]
The current kernel version of WSL2 is: `Linux 5.15.74.2-microsoft-standard-WSL2`, but [Intel's Thread Director Coming to Linux 5.18 to Fix Alder Lake Performance Issues](https://www.tomshardware.com/news/intel-thread-director-coming-to-linux-5-18). This means that WSL2 may have thread scheduling issues when running on Intel big.LITTLE CPUs.

Should/can I upgrade the kernel version? Or is there a way to run WSL2 only on P cores or E cores?

Thanks.

**Comment by 0xbadfca11** [comment]
It doesn't matter which virtual CPU the Linux kernel inside WSL schedules on.
It's the hypervisor's scheduler that actually decides which physical CPU to run on.
(You can see this by running `taskset 1 yes > /dev/null` and looking at `top 1` and taskmgr.exe.)

And those features are hidden by virtualization.

**Comment by dsyx** [comment]
@0xbadfca11 
But I'm a bit concerned about the hypervisor's scheduler scheduling my CPU-bound process in WSL2 onto the E core.

**Comment by tusharsnx** [comment]
similar issue is being tracked here #9121

**Comment by linzj** [comment]
I am running on Windows Server 2022, and in most case the heavy task simply schedule to the E cores.

**Comment by borjamunozf** [comment]
any new info about this?

**Reply by linzj** [reply]
I change the scheduler type of the hypervisor to 4, aka the root scheduler. Then the problem is solved.

**Reply by borjamunozf** [reply]
this applies to Windows 11? Or only Windows Server? The bcdedit does not show anything and it seems that root scheduler is the default (theorically)

We have huge performance issues in Windows 11 and WSL #11966

**Reply by linzj** [reply]
I have check my Windows 11 which is a client version OS. It defaulted to hypervisor 4. But my server version is not that case.

**Comment by dsyx** [comment]
I would recommend that the WSL config file provide a CPU affinity option, which would allow users to lock WSL to certain cores/threads instead of letting the scheduler run freely.
